app/src/utils/transfer.ts

```diff
@@ -43,3 +45,21 @@ export const formatDate = (date: string | Date): string => {
 export function feeToHuman(fees: Fees): string {
   return toHuman(fees.amount, fees.token).toFixed(10)
 }
+
+export async function lookupName(network: Network, address: string): Promise<string | null> {
+  switch (network) {
+    case Network.Ethereum: {
+      const provider = new ethers.CloudflareProvider()
+      try {
+        return provider.lookupAddress(address)
+      } catch {
+        // we are happy to fail silently here for now
+        return null
+      }
+    }
+    case Network.Polkadot: {
+      //todo(nuno)
+      return null
+    }
+  }
+}
```

Comment:
Can we move error handling to the hook `useLookupName`?
```suggestion
export async function lookupName(network: Network, address: string): Promise<string | null> {
  switch (network) {
    case Network.Ethereum: {
      const provider = new ethers.CloudflareProvider()
      return provider.lookupAddress(address)
    }
    case Network.Polkadot: {
      //todo(nuno)
      return null
    }
  }
}
```